src/lib/list/selection-list.spec.ts

```diff
@@ -0,0 +1,299 @@
+    let DOWN_EVENT: KeyboardEvent =
+      createKeyboardEvent('keydown', DOWN_ARROW, testListItem);
+    let options = selectionList.componentInstance.options;
+    let array = options.toArray();
+    let focusItem = array[2];
+    let manager = selectionList.componentInstance._keyManager;
+
+    focusItem.focus();
+    expect(manager.activeItemIndex).toEqual(2);
+
+    selectionList.componentInstance.keydown(DOWN_EVENT);
+
+    fixture.detectChanges();
+
+    expect(manager.activeItemIndex).toEqual(3);
+  });
+
+  it('should be able to customize checkbox position', () => {
+    let fixture = TestBed.createComponent(SelectionListWithCheckboxPositionAfter);
+    fixture.detectChanges();
+
+    let listItemContent = fixture.debugElement.query(By.css('.mat-list-item-content'));
+    expect(listItemContent.nativeElement.classList).toContain('mat-list-item-content-reverse');
+  });
+
+
+});
+
+@Component({template: `
+  <mat-selection-list id = "selection-list-1">
+    <md-list-option checkboxPosition = "before" disabled = "true">
+      Inbox (disabled selection-option)
+    </md-list-option>
+    <md-list-option id = "testSelect" checkboxPosition = "before" class="test-native-focus">
+      Starred
+    </md-list-option>
+    <md-list-option checkboxPosition = "before">
+      Sent Mail
+    </md-list-option>
+    <md-list-option checkboxPosition = "before">
+      Drafts
+    </md-list-option>
+  </mat-selection-list>`})
+class SelectionListWithListOptions {
+}
+
+@Component({template: `
+  <mat-selection-list id = "selection-list-2">
+    <md-list-option checkboxPosition = "after">
+      Inbox (disabled selection-option)
+    </md-list-option>
+    <md-list-option id = "testSelect" checkboxPosition = "after">
+      Starred
+    </md-list-option>
+    <md-list-option checkboxPosition = "after">
+      Sent Mail
+    </md-list-option>
+    <md-list-option checkboxPosition = "after">
+      Drafts
+    </md-list-option>
+  </mat-selection-list>`})
+class SelectionListWithCheckboxPositionAfter {
+}
+
+@Component({template: `
+  <mat-selection-list id = "selection-list-3" [disabled] = true>
+    <md-list-option checkboxPosition = "after">
+      Inbox (disabled selection-option)
+    </md-list-option>
+    <md-list-option id = "testSelect" checkboxPosition = "after">
+      Starred
+    </md-list-option>
+    <md-list-option checkboxPosition = "after">
+      Sent Mail
+    </md-list-option>
+    <md-list-option checkboxPosition = "after">
+      Drafts
+    </md-list-option>
+  </mat-selection-list>`})
+class SelectionListWithListDisabled {
+}
+
+@Component({template: `
+  <mat-selection-list id = "selection-list-4">
+    <md-list-option checkboxPosition = "after" class="test-focus" id="123">
+      Inbox
+    </md-list-option>
+  </mat-selection-list>`})
+class SelectionListWithOnlyOneOption {
+  isIE: boolean = true;
+  constructor(public platform: Platform) {
+    if (platform.SAFARI || platform.EDGE || platform.FIREFOX) {
```

Comment:
Changed `'(platform.SAFARI || platform.EDGE || platform.FIREFOX)'` to `'!platform.IS_TRIDENT'`